File: src/prod/src/client/InvokeQuorumLossProgressResult.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace Api;
using namespace ServiceModel;
using namespace Management::FaultAnalysisService;
using namespace Client;

InvokeQuorumLossProgressResult::InvokeQuorumLossProgressResult(
    shared_ptr<InvokeQuorumLossProgress> && progress)
    : invokeQuorumLossProgress_(move(progress))
{
}

shared_ptr<InvokeQuorumLossProgress> const & InvokeQuorumLossProgressResult::GetProgress()
{
    return invokeQuorumLossProgress_;
}

File: src/prod/src/managed/Api/src/System/Fabric/Result/InvokeQuorumLossProgressResult.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System.Fabric.Interop;

    internal sealed class InvokeQuorumLossProgressResult : NativeClient.IFabricPartitionQuorumLossProgressResult, IDisposable
    {
        private readonly IntPtr nativeResult;
        private readonly PinCollection pinCollection;
        private bool disposed;

        public InvokeQuorumLossProgressResult(PartitionQuorumLossProgress progress)
        {
            this.pinCollection = new PinCollection();
            this.nativeResult = progress.ToNative(pinCollection);
        }

        public IntPtr get_Progress()
        {
            if (this.disposed)
            {
                throw new ObjectDisposedException("InvokeQuorumLossProgressResult has been disposed");
            }

            return this.nativeResult;
        }

        public void Dispose()
        {
            this.pinCollection.Dispose();
            this.disposed = true;
        }
    }
}